### PR TITLE
stat: Add sequential area output to stat -liberty

### PR DESCRIPTION
**Pull request by QuantamHD**
Checks to see if a cell is of type ff in the liberty, and keeps track of an additional area value.

```
   Chip area for module '\addr': 92.280720
   Sequential area for module '\addr': 38.814720
```

### CODE REVIEW

**Review by mmicko — COMMENTED** [review]
Code looks fine, and I also checked liberty files documentation to check if there is anything missing, but all looks fine.
However, I would suggest renames to better represent structures.

cell_properties - > cell_area (so back as it was)
cell_data_t -> cell_area_t
is_flip_flop -> is_sequential

**Comment by QuantamHD** [comment]
Done.

**Review by mmicko — APPROVED** [review]
Looks good to me now

File: passes/cmds/stat.cc [hunk 1]
```diff
@@ -244,6 +256,7 @@ struct statdata_t
 		if (area != 0) {
 			log("\n");
 			log("   Chip area for %smodule '%s': %f\n", (top_mod) ? "top " : "", mod_name.c_str(), area);
+			log("   Sequential area for %smodule '%s': %f\n", (top_mod) ? "top " : "", mod_name.c_str(), sequential_area);
```
Comment (nakengelhardt) [inline passes/cmds/stat.cc:259]: I think the text could make it clearer that these two are not cumulative, maybe something like this:
```suggestion
			log("     of which used for sequential logic: %f (%.2f%%)\n", sequential_area, 100.0*sequential_area/area);
```

**Pull request merged**